pgjdbc/src/main/java/org/postgresql/jdbc/TimestampUtils.java

```diff
@@ -1556,27 +1562,30 @@ private static long toPgSecs(long secs) {
   /**
    * Converts the SQL Date to binary representation for {@link Oid#DATE}.
    *
-   * @param tz The timezone used.
+   * @param tz The timezone used. If <code>null</code>, then no timezone offset will be applied to the date.
    * @param bytes The binary encoded date value.
    * @param value value
    * @throws PSQLException If binary format could not be parsed.
    */
   public void toBinDate(@Nullable TimeZone tz, byte[] bytes, Date value) throws PSQLException {
-    long millis = value.getTime();
-
-    if (tz == null) {
-      tz = getDefaultTz();
+    if (tz != null) {
+      long millis = value.getTime();
+      value = new Date(millis + tz.getOffset(millis));
     }
-    // It "getOffset" is UNTESTED
-    // See org.postgresql.jdbc.AbstractJdbc2Statement.setDate(int, java.sql.Date,
-    // java.util.Calendar)
-    // The problem is we typically do not know for sure what is the exact required date/timestamp
-    // type
-    // Thus pgjdbc sticks to text transfer.
-    millis += tz.getOffset(millis);
 
-    long secs = toPgSecs(millis / 1000);
```

Comment:
The method toPgSecs is now unused. I would still suggest to make sure that toPgSecs and toJavaSecs are symmetric and Gruß method should possibly use it. We should add a test for that.

Reply:
I'll take a stab at that, but it'll probably be during the weekend.